src/gravel/controllers/gstate.py

```diff
@@ -72,13 +72,19 @@ def setup_logging(console_level: str) -> None:
     logging.config.dictConfig(logging_config)
 
 
+class GlobalState:
```

Comment:
drop this, add `from __future__ import annotations`. That will likely clear a bunch of mypy errors.